wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php

```diff
@@ -85,19 +90,42 @@ function webonary_searchform() {
 			$indexed->language_name = $dictionary->mainLanguage->title ?? $dictionary->mainLanguage->lang;
 			$indexed->totalIndexed = $dictionary->mainLanguage->entriesCount ?? 0;
 			$arrIndexed[] = $indexed;
-			foreach($dictionary->reversalLanguages as $index => $reversal)
-			{
-				$indexed = new stdClass();
-				$indexed->language_name = $reversal->title ?? $dictionary->reversal->lang;
-				$indexed->totalIndexed = $reversal->entriesCount ?? 0;
-				$arrIndexed[] = $indexed;
+
+			if (count($dictionary->reversalLanguages)) {
+				$selected = ($dictionary->mainLanguage->lang === $selected_language) ? ' selected' : '';
+				$language_dropdown_options .= "<option value='" . $dictionary->mainLanguage->lang . "'" . $selected . ">" . $indexed->language_name . "</option>";	
+				foreach($dictionary->reversalLanguages as $index => $reversal)
+				{
+					$indexed = new stdClass();
+					$indexed->language_name = $reversal->title ?? $reversal->lang;
+					$indexed->totalIndexed = $reversal->entriesCount ?? 0;
+					$arrIndexed[] = $indexed;
+	
+					// set up languages dropdown options
+					$selected = ($reversal->lang === $selected_language) ? ' selected' : '';
+					$language_dropdown_options .= "<option value='" . $reversal->lang . "'" . $selected . ">" . $indexed->language_name . "</option>";	
+				}	
 			}
 
 			$lastEditDate = $dictionary->updatedAt;
 		}
 	}
 	else
 	{
+		//$catalog_terms = get_terms('sil_writing_systems');
+		$arrLanguages = Webonary_Configuration::get_LanguageCodes();
+		$arrVernacularLanguage = Webonary_Configuration::get_LanguageCodes(get_option('languagecode'));
```

Comment:
This line should be
`$arrVernacularLanguage = Webonary_Configuration::get_LanguageCodes(get_option('languagecode'))[0]`
That way arrVernacularLanguage won't be an array.

Reply:
Done